### PR TITLE
feat: Add folder_ids to the GCP spec

### PR DESCRIPTION
**Pull request by shimonp21**

### CODE REVIEW

File: plugins/source/gcp/client/client.go [hunk 1]
```diff
@@ -46,6 +50,12 @@ func (c *Client) Logger() *zerolog.Logger {
 }
 
 func New(ctx context.Context, logger zerolog.Logger, s specs.Source) (schema.ClientMeta, error) {
+	// We use a timeout to make sure that this method doesn't get stuck indefinitely.
+	ctx, cancel := context.WithTimeout(ctx, time.Minute)
```
Comment (yevgenypats) [inline plugins/source/gcp/client/client.go:54]: I think a minute is quite low. For example we saw users with more than 80k projects. 
Also I think there is no need for calling `cancel`

File: plugins/source/gcp/client/spec.go [hunk 2]
```diff
@@ -4,4 +4,6 @@ package client
 type Spec struct {
 	ProjectIDs            []string `json:"project_ids"`
 	ServiceAccountKeyJSON string   `json:"service_account_key_json"`
+	FolderIDs             []string `json:"folder_ids"`
+	FolderRecursionDepth  int      `json:"folder_recursion_depth"`
```
Comment (yevgenypats) [inline plugins/source/gcp/client/spec.go:8]: let's setup default for `FolderRecursionDepth` via `SetDefault` method

**Review by yevgenypats — CHANGES_REQUESTED** [review]
Looks good. few comments

File: plugins/source/gcp/client/client.go (hunk 1 above)
Comment (shimonp21) [reply plugins/source/gcp/client/client.go]: Removed. Will be added at the SDK level if required later.

File: plugins/source/gcp/client/spec.go (hunk 2 above)
Comment (shimonp21) [reply plugins/source/gcp/client/spec.go]: Changed to *int.

**Review by yevgenypats — APPROVED** [review]

File: plugins/source/gcp/client/client.go [hunk 3]
```diff
@@ -60,6 +68,13 @@ func New(ctx context.Context, logger zerolog.Logger, s specs.Source) (schema.Cli
 
 	serviceAccountKeyJSON := []byte(gcpSpec.ServiceAccountKeyJSON)
 
+	var folderRecursionDepth int
+	if gcpSpec.FolderRecursionDepth == nil {
```
Comment (yevgenypats) [inline plugins/source/gcp/client/client.go:72]: This should go as a method function though. i.e `gcpSpec.SetDefaults()`

**Pull request merged**